src/document.rs

```diff
@@ -314,14 +328,18 @@ impl Document {
                 .unwrap_or(default_excerpt_separator);
 
             let excerpt = if let Some(excerpt_str) = excerpt_attr {
-                excerpt_str
+                excerpt_str.to_string()
             } else if excerpt_separator.is_empty() {
-                ""
+                String::new()
             } else {
-                self.content.split(excerpt_separator).next().unwrap_or(&self.content)
+                self.extract_references() +
```

Comment:
and personally I'd like references to be added to the end of excerpt (feel free to ignore me here though)